src/Microsoft.AspNet.OData.Shared/Builder/ActionConfiguration.cs

```diff
@@ -233,6 +233,47 @@ public ActionConfiguration SetBindingParameter(string name, IEdmTypeConfiguratio
             return this;
         }
 
+        /// <summary>
+        /// Adds subtypes to the list of derived type constraints for the return type.
+        /// </summary>
+        /// <param name="subtypes">The subtypes for which the constraint needs to be added.</param>
+        /// <returns>Updated configuration object.</returns>
+        public ActionConfiguration HasDerivedTypeConstraintsForReturnType(params Type[] subtypes)
+        {
+            if (ReturnType == null)
+            {
+                throw Error.InvalidOperation(SRResources.ReturnTypeOfOperationNotSpecified);
+            }
+
+            if (ReturnTypeConstraints == null)
+            {
+                ReturnTypeConstraints = new DerivedTypeConstraintConfiguration(ReturnType.ClrType);
+            }
+
+            ReturnTypeConstraints.AddConstraints(subtypes);
```

Comment:
why don't you move all these codes into base class OperationConfiguration?

Reply:
To avoid the cast while using the fluent APIs. 

---
In reply to: [401813541](https://github.com/OData/WebApi/pull/1977#discussion_r401813541) [](ancestors = 401813541)